pkg/scheduler/framework/plugins/podtopologyspread/scoring_test.go

```diff
@@ -147,6 +147,41 @@ func TestPreScoreStateEmptyNodes(t *testing.T) {
 				TopologyNormalizingWeight: []float64{topologyNormalizingWeight(1), topologyNormalizingWeight(1)},
 			},
 		},
+		{
+			name: "system defaults constraints and nodes don't have label zone",
+			pod:  st.MakePod().Name("p").Namespace("default").Label("foo", "tar").Label("baz", "sup").OwnerReference("rs1", appsv1.SchemeGroupVersion.WithKind("ReplicaSet")).Obj(),
+			config: config.PodTopologySpreadArgs{
+				DefaultingType: config.SystemDefaulting,
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label(v1.LabelHostname, "node-a").Label(v1.LabelTopologyZone, "zone1").Obj(),
```

Comment:
Keep in mind that, in this scenario, there are 2 zones: "zone1" and "".
It's worth having a test for `Score` for no zones to see how the scores behave. In that scenario, `SystemDefaulting` should behave as `ListDefaulting` that only has a constraint for the node topology (or very close to it). I'm not sure if this change is achieving that.
Or perhaps `topoSize` is achieving it. Please include a test for the two defaultings described so we can compare.

Reply:
done, I have written a test, but the test failed :(, the two cases' score is no even very close to each other.

This implementation does not behave as we wanted.  the last result will penalize nodes with all required topology keys presented and favor without. When using system default constraints, the node which has zone labels will get more pod-matched constraint selector and result in less score.